action/action.rb

```diff
@@ -101,7 +101,13 @@ def report_offense?(offense, change_ranges:)
                  else
                    event.pull_request.base.sha
                  end
-  annotations = generate_annotations(compare_sha: previous_sha)
+  ActionUtils.debug "Using this as previous sha: #{previous_sha}"
+  annotations = if previous_sha == '0000000000000000000000000000000000000000'
```

Comment:
I was feeling a similar thing about the relative riskiness of this check.

I quickly started zooming out a bit to how this particular problem might show up in other places.

Do we want to silently return no annotations in this scenario or do we want to throw an error?  This seems like an exceptional circumstance where use an error to indicate the issue (maybe even checking the presence of the sha with [`git cat-file -e {sha}`](https://git-scm.com/docs/git-cat-file)?) seems reasonable.

That codifies the underlying issue, and then we can prevent the issue in our repo (and same for other folks using this action) by filtering out problematic pushes?  

In general, does saying something went just fine when we didn't do any of the work feel potentially misleading?

(if this broader conversation needs to wait in order to get a quick fix out, I'm good with that too)

Reply:
These are all great points. Here's what I'm thinking:

1. Validations could be wider, I'll work on that
2. It _does_ feel weird that it would be listed the same as a successful check when this happens. I'll look into seeing what we can do here to change that.